### PR TITLE
Add native delayed delivery API to kombu

### PR DESCRIPTION
**Pull request by thedrow**
This PR adds the native delayed delivery API to kombu so that it can be used without Celery if required.
The PR includes an example on how to do so.

### CODE REVIEW

**Review by Nusnus — CHANGES_REQUESTED** [review]
1. Add official documentation.
2. Make sure the CI passes.
3. Make sure you're rebased on main.

Also, I'm not sure about `kombu.delayed...`. Shouldn't it be in the RabbitMQ transport layer?

This feature is relevant only for RabbitMQ so it should be in the matching place in the code IMHO.

**Comment by thedrow** [comment]
I thought about it and I prefer to keep the API in a separate module.
I included the example in our list of examples and added a short description which is enough documentation for most users.

The pre-commit check is failing for unrelated reasons.

**Review by Nusnus — DISMISSED** [review]
> I thought about it and I prefer to keep the API in a separate module.
> I included the example in our list of examples and added a short description which is enough documentation for most users.
> 
> The pre-commit check is failing for unrelated reasons.

There's no problem leaving it in a separate module, but correct me if I'm wrong, this is a transport layer feature, so it should be in the transport.

This is a cosmetic change.

Apart from that, please add the version string to the docs with v5.5 (Current 5.4.2).

We don't expect any patch release soon so the next minor release will include your new feature.

Besides LGTM, thank you!

File: docs/reference/kombu.native_delayed_delivery.rst
Comment (Nusnus) [inline docs/reference/kombu.native_delayed_delivery.rst:1]: Please add version string

File: docs/reference/index.rst [hunk 1]
```diff
@@ -28,6 +28,7 @@ Kombu Core
     kombu.abstract
     kombu.resource
     kombu.serialization
+    kombu.native_delayed_delivery
```
[the remark this answers is not in the document]
Comment (thedrow) [reply docs/reference/index.rst:31]: But that's where the transports are. This module is not a transport.

**Review by auvipy — COMMENTED** [review]
yes, Native Delayed Delivery API is to Transport layer, so logically, this should be inside the transport

**Comment by Nusnus** [comment]
CleanShot 2024-10-12 at 22 15 57@2x

Thank you for explaining @thedrow , but next time please just ping me here so I’ll review the fixes and approve the PR instead.
I need to confirm my review comments are resolved, or it loses its entire point 🤷‍♂️

Please pay attention next time, thank you.

**Pull request merged**